File: platforms/android/sample/app/src/main/java/io/nativeblocks/sample/ui/frame/FrameRoute.kt
package io.nativeblocks.sample.ui.frame

import io.nativeblocks.runtime.api.provider.model.NativeFrameRouteModel

data class FrameRoute(
    val name: String,
    val route: String,
)

internal fun NativeFrameRouteModel.toFrameRoute(): FrameRoute? {
    val route = route ?: return null
    return FrameRoute(name = name ?: route, route = route)
}

File: platforms/ios/sample/sample/ui/frame/FrameRoute.swift
import Foundation
import NativeblocksRuntime

struct FrameRoute: Hashable {
    let name: String
    let route: String
}

extension NativeFrameRouteModel {

    func toFrameRoute() -> FrameRoute? {
        guard let route else { return nil }
        return FrameRoute(name: name ?? route, route: route)
    }
}
